extension/src/WorkspaceFunctions.ts

```diff
@@ -13,41 +13,30 @@ const invalidChars = [":", "/", "\\", "?", "<", ">", "*", "|", "\""];
 
 // private static gXLFFilepath: string;
 export async function openAlFileFromXliffTokens(tokens: XliffIdToken[]) {
-
-    let alFiles = await getAlFilesFromCurrentWorkspace();
-    for (let index = 0; index < alFiles.length; index++) {
-        const alFile = alFiles[index];
-        let fileContent: string = fs.readFileSync(alFile.fsPath, 'UTF8');
-        let obj = ALObject.getALObject(fileContent, false, alFile.fsPath);
-        if (obj) {
-
-            if (ALObjectType[obj.objectType].toLowerCase() === tokens[0].type.toLowerCase() && obj.objectName.toLowerCase() === tokens[0].name.toLowerCase()) {
-                // found our file!
-                obj = ALObject.getALObject(fileContent, true, alFile.fsPath);
-                if (!obj) {
-                    throw new Error(`Could not parse the file '${alFile.fsPath}'`);
-                }
-                let xliffToSearchFor = XliffIdToken.getXliffId(tokens).toLowerCase();
-                let mlObjects = obj.getAllMultiLanguageObjects({ onlyForTranslation: true });
-                let mlObject = mlObjects.filter(x => x.xliffId().toLowerCase() === xliffToSearchFor);
-                if (mlObject.length !== 1) {
-                    throw new Error(`No code line found in file '${alFile.fsPath}' matching '${XliffIdToken.getXliffIdWithNames(tokens)}'`);
-                }
-                DocumentFunctions.openTextFileWithSelectionOnLineNo(alFile.fsPath, mlObject[0].startLineIndex);
-                break;
-            }
-        }
-
+    const alObjects = await getAlObjectsFromCurrentWorkspace(false);
+    let obj = alObjects.filter(x => ALObjectType[x.objectType].toLowerCase() === tokens[0].type.toLowerCase() && x.objectName.toLowerCase() === tokens[0].name.toLowerCase())[0];
+    if (!obj) {
+        throw new Error(`Could not find any object matching '${XliffIdToken.getXliffIdWithNames(tokens)}'`);
+    }
+    // found our object, load complete object from file
+    obj.loadObject();
+
+    let xliffToSearchFor = XliffIdToken.getXliffId(tokens).toLowerCase();
+    let mlObjects = obj.getAllMultiLanguageObjects({ onlyForTranslation: true });
+    let mlObject = mlObjects.filter(x => x.xliffId().toLowerCase() === xliffToSearchFor);
```

Comment:
Since mlObjects isn't used for anything else.
```suggestion
    let mlObject = obj.getAllMultiLanguageObjects({ onlyForTranslation: true }).filter(x => x.xliffId().toLowerCase() === xliffToSearchFor);
```

Reply:
I don't like the long oneliners because of what they do to debugging... I tend to break them down into local variables when I need to debug it, so why not do it like that to begin with...? 
I just don't see the value in super long combined oneliners, except it might be cool that it's even possible.😉
Guess it's a matter of taste...

Reply:
Agreed. The debugging part is usually what decides it for me, because - as you said, you just now that you're going to reverse it when you're stuck debugging :D